### PR TITLE
Refactoring all instances of "algo" to "arrows"

### PR DESCRIPTION
**Pull request by anthony-barricelli**
See #761

### CODE REVIEW

**Comment by mleotta** [comment]
Jenkins test this please

**Comment by mleotta** [comment]
@linus-sherrill @kfieldho @dstoup @mattdawkins 

I suggested this change to @anthony-barricelli.  I think it's the right thing to do, but it will likely require updates in other projects like Viame, Squad-X, TeleSculptor, etc.  Basically all the arrows libraries are named "kwiver_algo_*something*", which I think was left over from the naming conventions in MAP-Tk.  I think "kwiver_arrows_*something*" makes more sense and matches both the namespace and directory layout of these files.  Assuming all the CI builds pass, are there any objections to making this change?  It's a simple change, but touches a lot of files.

**Comment by mleotta** [comment]
@anthony-barricelli from the CI builds it looks like there are still places (e.g. outside of arrows) that reference the old library names.

**Comment by mleotta** [comment]
Jenkins test this please

**Comment by dstoup** [comment]
The change itself looks like a good thing. I am happy provided the dashboards are.

File: arrows/vxl/polygon.h [hunk 1]
```diff
@@ -33,10 +33,10 @@
  * \brief vxl polygon conversion interface
  */
 
-#ifndef KWIVER_ALGORITHM_VXL_POLYGON_H
-#define KWIVER_ALGORITHM_VXL_POLYGON_H
+#ifndef KWIVER_ARROWSRITHM_VXL_POLYGON_H
+#define KWIVER_ARROWSRITHM_VXL_POLYGON_H
```
Comment (mattdawkins) [inline arrows/vxl/polygon.h]: looks like a bad copy paste or find replace here

File: arrows/vxl/polygon.h [hunk 2]
```diff
@@ -71,9 +71,9 @@ std::shared_ptr< vgl_polygon< double > > vital_to_vxl( kwiver::vital::polygon_sp
...
  * @throws std::out_of_range if the input polygon does not have any sheets/contours.
  */
-KWIVER_ALGO_VXL_EXPORT
+KWIVER_ARROWS_VXL_EXPORT
 kwiver::vital::polygon_sptr vxl_to_vital( const vgl_polygon< double >& poly );
 
 } } } // end namespace
 
-#endif // KWIVER_ALGORITHM_VXL_POLYGON_H
+#endif // KWIVER_ARROWSRITHM_VXL_POLYGON_H
```
Comment (mattdawkins) [inline arrows/vxl/polygon.h]: looks like a bad copy paste or find replace here

**Comment by linus-sherrill** [comment]
This looks like the right thing to do, but landing it will require rework of most other outstanding branches.

**Pull request closed without merging**